goosebit/api/rollouts/routes.py

```diff
@@ -0,0 +1,51 @@
+from fastapi import APIRouter, Security
+from fastapi.requests import Request
+
+from goosebit.api.responses import StatusResponse
+from goosebit.auth import validate_user_permissions
+from goosebit.models import Rollout
+from goosebit.permissions import Permissions
+
+from .requests import RolloutsDeleteRequest, RolloutsPatchRequest, RolloutsPutRequest
+from .responses import RolloutsPutResponse, RolloutsResponse
+
+router = APIRouter(prefix="/rollouts", tags=["rollouts"])
+
+
+@router.get(
+    "",
+    dependencies=[Security(validate_user_permissions, scopes=[Permissions.ROLLOUT.READ])],
+)
+async def rollouts_get(_: Request) -> RolloutsResponse:
+    return await RolloutsResponse.convert(await Rollout.all().prefetch_related("firmware"))
+
+
+@router.put(
```

Comment:
Looking at the web docs, it seems like I misunderstood the first time.  The way I understand it now is -
POST - create a resource 
PUT - overwrite an existing resource
PATCH - update and existing resource

I'll get this fixed up tomorrow.

Another thing on this, I think we may want to rename the `post` function in JS to `post_request` as that was the name I went for (since `delete` is technically a reserved keyword)

Reply:
Let's do it in a fresh release. I also have a few things for a PR but wanted to wait for this big one to hit master first.